spring-integration-core/src/main/java/org/springframework/integration/handler/LambdaMessageProcessor.java

```diff
@@ -170,4 +185,41 @@ else if (Map.class.isAssignableFrom(parameterType)) {
 		return args;
 	}
 
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	private Object invokeMethod(Object[] args) throws InvocationTargetException, IllegalAccessException {
+		if (this.target instanceof Function function) {
+			return function.apply(args[0]);
+		}
+		else if (this.target instanceof GenericSelector selector) {
+			return selector.accept(args[0]);
+		}
+		else if (this.target instanceof GenericTransformer transformer) {
+			return transformer.transform(args[0]);
+		}
+		else if (this.target instanceof GenericHandler handler) {
+			return handler.handle(args[0], (MessageHeaders) args[1]);
+		}
+		else {
+			return this.method.invoke(this.target, args);
+		}
+
+		/* TODO when preview features are available in the next Java version
```

Comment:
This might need a NOSONAR (TODO).

Reply:
I think that's OK: will give us an extra reminder to not forget to fix this eventually 😄

Reply:
Either way our Sonar instance does not support Java 17 yet: https://stackoverflow.com/questions/71528775/migrate-sonarqube-to-support-java-17